test/integration/auth/svcaccttoken_test.go

```diff
@@ -63,103 +64,162 @@ func TestServiceAccountTokenCreate(t *testing.T) {
 		t.Fatalf("err: %v", err)
 	}
 
-	sa := &v1.ServiceAccount{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "test",
-			Namespace: "default",
-		},
-	}
+	var (
+		sa = &v1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test",
+				Namespace: "default",
+			},
+		}
+		pod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-pod",
+				Namespace: sa.Namespace,
+			},
+			Spec: v1.PodSpec{
+				ServiceAccountName: sa.Name,
+				Containers:         []v1.Container{{Name: "test-container", Image: "nginx"}},
+			},
+		}
+		secret = &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-secret",
+				Namespace: sa.Namespace,
+			},
+		}
 
-	tr1 := &authenticationv1.TokenRequest{
-		Spec: authenticationv1.TokenRequestSpec{
-			Audiences: []string{"aud"},
-		},
-	}
+		one      = int64(1)
+		wrongUID = types.UID("wrong")
+		noUID    = types.UID("")
+	)
 
-	_, err = cs.CoreV1().ServiceAccounts(sa.Namespace).Create(sa)
-	if err != nil {
-		t.Fatalf("err: %v", err)
-	}
-	tr1, err = cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(sa.Name, tr1)
-	if err != nil {
-		t.Fatalf("err: %v", err)
-	}
+	t.Run("bound to service account", func(t *testing.T) {
+		treq := &authenticationv1.TokenRequest{
+			Spec: authenticationv1.TokenRequestSpec{
+				Audiences:         []string{"api"},
+				ExpirationSeconds: &one,
+			},
+		}
 
-	checkPayload(t, tr1.Status.Token, `"system:serviceaccount:default:test"`, "sub")
-	checkPayload(t, tr1.Status.Token, `["aud"]`, "aud")
-	checkPayload(t, tr1.Status.Token, "null", "kubernetes.io", "pod")
-	checkPayload(t, tr1.Status.Token, "null", "kubernetes.io", "secret")
-	checkPayload(t, tr1.Status.Token, `"default"`, "kubernetes.io", "namespace")
-	checkPayload(t, tr1.Status.Token, `"test"`, "kubernetes.io", "serviceaccount", "name")
-
-	pod := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "test-pod",
-			Namespace: sa.Namespace,
-		},
-		Spec: v1.PodSpec{
-			ServiceAccountName: sa.Name,
-			Containers:         []v1.Container{{Name: "test-container", Image: "nginx"}},
-		},
-	}
-	_, err = cs.CoreV1().Pods(pod.Namespace).Create(pod)
-	if err != nil {
-		t.Fatalf("err: %v", err)
-	}
+		if treq, err := cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(sa.Name, treq); err == nil {
```

Comment:
Call the response `tres` or `tresp` or something.

Reply:
I know they're the same type, and shadowing prevents you from overwriting the first one, but it confuses my small brain.

Reply:
Fixed